metadata-ingestion/src/datahub/ingestion/source/sql/postgres.py

```diff
@@ -67,10 +126,93 @@ class PostgresSource(SQLAlchemySource):
     - Table, row, and column statistics via optional SQL profiling
     """
 
-    def __init__(self, config, ctx):
+    def __init__(self, config: PostgresConfig, ctx: PipelineContext):
         super().__init__(config, ctx, "postgres")
 
     @classmethod
     def create(cls, config_dict, ctx):
         config = PostgresConfig.parse_obj(config_dict)
         return cls(config, ctx)
+
+    def get_workunits(self) -> Iterable[Union[MetadataWorkUnit, SqlWorkUnit]]:
+        yield from super().get_workunits()
+
+        if self.config.include_view_lineage:  # type: ignore
+            yield from self._get_view_lineage_workunits()
+
+    def _get_view_lineage_workunits(self) -> Iterable[MetadataWorkUnit]:
+        url = self.config.get_sql_alchemy_url()
+        engine = create_engine(url, **self.config.options)
+
+        data: List[ViewLineageEntry] = []
+
+        with engine.connect() as conn:
+            results: CursorResult = conn.execute(VIEW_LINEAGE_QUERY)
+            if results.returns_rows is False:
+                return None
+
+            for row in results:
+                data.append(ViewLineageEntry.parse_obj(row))
+
+        if len(data) == 0:
+            return None
+
+        lineage_elements: Dict[str, List[str]] = {}
+        # Loop over the lineages in the JSON data.
+        for lineage in data:
+            if not self.config.view_pattern.allowed(lineage.dependent_view):
+                self.report.report_dropped(
+                    f"{lineage.dependent_schema}.{lineage.dependent_view}"
+                )
+                continue
+
+            if not self.config.schema_pattern.allowed(lineage.dependent_schema):
+                self.report.report_dropped(
+                    f"{lineage.dependent_schema}.{lineage.dependent_view}"
+                )
+                continue
+
+            # Check if the dependent view + source schema already exists in the dictionary, if not create a new entry.
+            # Use ':::::' to join as it is unlikely to be part of a schema or view name.
+            key = ":::::".join([lineage.dependent_view, lineage.dependent_schema])
+            if key not in lineage_elements:
+                lineage_elements[key] = []
+
+            # Append the source table to the list.
+            lineage_elements[key].append(
+                mce_builder.make_dataset_urn(
+                    "postgres",
+                    self.config.get_identifier(  # type: ignore
```

Comment:
why was the type ignore required here?

Reply:
I found it necessary because mypy returns this error 

```
src/datahub/ingestion/source/sql/postgres.py:203: error: "SQLAlchemyConfig" has no attribute "get_identifier"  [attr-defined]
```

Because `self.config` is `PostgresConfig` which extends `SQLAlchemyConfig`, but mypy is not aware. I wasn't sure how to best fix it, I was hoping someone here might have a better way.

Reply:
got it - you can "narrow" the type like this

```diff
@@ -129,6 +129,8 @@ class PostgresSource(SQLAlchemySource):
     - Table, row, and column statistics via optional SQL profiling
     """

+    config: PostgresConfig
+
     def __init__(self, config: PostgresConfig, ctx: PipelineContext):
         super().__init__(config, ctx, "postgres")
```